components/org.wso2.carbon.identity.authz.valve/src/main/java/org/wso2/carbon/identity/authz/valve/AuthorizationValve.java

```diff
@@ -78,34 +76,42 @@ public void invoke(Request request, Response response) throws IOException, Servl
-            try {
-                AuthorizationResult authorizationResult = authorizationManager.authorize(authorizationContext);
-                if (authorizationResult.getAuthorizationStatus().equals(AuthorizationStatus.GRANT)) {
-                    getNext().invoke(request, response);
-                } else {
-                    handleErrorResponse(authenticationContext, response, HttpServletResponse.SC_FORBIDDEN);
+
+            if (!isUserEmpty(authenticationContext)) {
+                if (resourceConfig != null && StringUtils.isNotEmpty(resourceConfig.getPermissions())) {
+                    authorizationContext.setPermissionString(resourceConfig.getPermissions());
+                }
+                if (resourceConfig != null && CollectionUtils.isNotEmpty(resourceConfig.getScopes())) {
+                    authorizationContext.setRequiredScopes(resourceConfig.getScopes());
+                }
+                String contextPath = request.getContextPath();
+                String httpMethod = request.getMethod();
+                authorizationContext.setContext(contextPath);
+                authorizationContext.setHttpMethods(httpMethod);
+                authorizationContext.setUser(authenticationContext.getUser());
+                authorizationContext.addParameter(OAUTH2_ALLOWED_SCOPES, authenticationContext.getParameter(OAUTH2_ALLOWED_SCOPES));
+                authorizationContext.addParameter(OAUTH2_VALIDATE_SCOPE, authenticationContext.getParameter(OAUTH2_VALIDATE_SCOPE));
+
+                String tenantDomainFromURLMapping = Utils.getTenantDomainFromURLMapping(request);
+                authorizationContext.setTenantDomainFromURLMapping(tenantDomainFromURLMapping);
+                List<AuthorizationManager> authorizationManagerList = AuthorizationValveServiceHolder.getInstance()
+                        .getAuthorizationManagerList();
+                AuthorizationManager authorizationManager = HandlerManager.getInstance()
+                        .getFirstPriorityHandler(authorizationManagerList, true);
+                try {
+                    AuthorizationResult authorizationResult = authorizationManager.authorize(authorizationContext);
+                    if (authorizationResult.getAuthorizationStatus().equals(AuthorizationStatus.GRANT)) {
+                        getNext().invoke(request, response);
+                    } else {
+                        handleErrorResponse(authenticationContext, response, HttpServletResponse.SC_FORBIDDEN);
+                    }
+                } catch (AuthzServiceServerException e) {
+                    handleErrorResponse(authenticationContext, response, HttpServletResponse.SC_BAD_REQUEST);
                 }
-            } catch (AuthzServiceServerException e) {
-                handleErrorResponse(authenticationContext, response, HttpServletResponse.SC_BAD_REQUEST);
+            } else {
+                getNext().invoke(request, response);
             }
+
```

Comment:
Remove redundant line

Reply:
Addressed with commit 6eceab04227033c7cfc1479f372ff2684d222ad2 in https://github.com/wso2-extensions/identity-carbon-auth-rest/pull/162